ibis/expr/types/structs.py

```diff
@@ -205,7 +205,7 @@ def __getitem__(self, name: str) -> ir.Value:
         KeyError: 'foo_bar'
         """
         if name not in self.names:
-            raise KeyError(name)
+            raise FieldNotFoundError(self, name, self.names)
```

Comment:
I think it's OK to be breaking here and not raise a KeyError anymore?

Reply:
KeyError seems semantically a little wrong. I think it should be for collections with a dynamic set of keys. But structs have a static set of keys, so FieldNotFoundError, as a subclass of AttributeError, seems better to me.